proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+        if (currChar == '*')
+          commentLevel++;
+      } else if (currChar == '*') {
+        while (currChar == '*') {
+          getChar();
+          unEscapedAppend();
+          if (currChar == '/')
+            commentLevel--;
+        }
+      } else if (currInt == EOF_CHAR) {
+        prepro.lexicalThrow("Missing end of comment");
+      }
+    }
+    prepro.setDoingComment(false);
+    getChar();
+    return makeToken(PreprocessorParser.COMMENT);
+  }
+
+  ProToken singleLineComment() throws IOException {
+    LOGGER.trace("Entering singleLineComment()");
+
+    // Single line comments are treated just like regular comments,
+    // everything till end of line is considered comment - no escape
+    // character to look after
+    append(); // currChar=='/'
+
+    while (true) {
+      getChar();
+      unEscapedAppend();
+      if (currChar == '\r' || currChar == '\n' || currInt == EOF_CHAR) {
+        return makeToken(PreprocessorParser.COMMENT);
+      }
+    }
+  }
+
+  ProToken quotedString() throws IOException {
+    LOGGER.trace("Entering quotedString()");
+
+    // Inside quoted strings (string constants) we preserve
+    // the source code's original text - we don't discard
+    // escape characters.
+    // The preprocessor *does* expand macros inside strings.
+    for (;;) {
+      if (currInt == EOF_CHAR)
+        prepro.lexicalThrow("Unmatched quote");
+      unEscapedAppend();
+      if (currInt == currStringType && !prepro.isEscapeCurrent()) {
+        getChar();
+        if (currInt == currStringType) { // quoted quote
+          unEscapedAppend();
+        } else {
+          break; // close quote
+        }
+      }
+      getChar();
+    }
+
+    if (currChar == ':') {
+      boolean isStringAttributes = false;
+      // Preserve the colon before calling getChar,
+      // in case it belongs in the next token.
+      preserveCurrent();
+      String theText = ":";
+      for_loop : for (;;) {
+        getChar();
+        switch (currChar) {
+          case 'r':
+          case 'l':
+          case 'c':
+          case 't':
+          case 'u':
+          case 'x':
+          case '0':
+          case '1':
+          case '2':
+          case '3':
+          case '4':
+          case '5':
+          case '6':
+          case '7':
+          case '8':
+          case '9':
+            theText += (char) currInt;
+            isStringAttributes = true;
+            break;
+          default:
+            break for_loop;
+        }
+      }
+      // either string attributes, or the preserved colon
+      // goes into the next token.
+      if (isStringAttributes) {
+        append(theText);
+        preserveDrop();
+      }
+    } // currChar==':'
+
+    return makeToken(PreprocessorParser.QSTRING);
+  }
+
+  ProToken digitStart() throws IOException {
+    LOGGER.trace("Entering digitStart()");
+
+    int ttype = PreprocessorParser.NUMBER;
+    for_loop : for (;;) {
+      switch (currChar) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of switch cases from 47 to at most 30. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1479)